deepchem/data/data_loader.py

```diff
@@ -1051,6 +1050,226 @@ def _add_sequence(sequences: np.ndarray,
     return DiskDataset.create_dataset(shard_generator(), data_dir)
 
 
+def _fastq_load_files(input_files: List[str],
+                      shard_size: Optional[int] = 4096) -> Iterator:
+  """Load data as Iterator.
+
+  Parameters
+  ----------
+  input_files: List[str]
+    List of fastq filenames.
+  shard_size: int, optional (default 4096)
+    Chunksize for reading fastq files.
+
+  Yields
+  -------
+  Iterator
+    Generator which yields the data which is the same shard size.
+  """
+  shard_num = 0
+  for input_file in input_files:
+    logger.info("About to start loading fastq from %s." % input_file)
+    # Open index file
+    with open(input_file, 'r') as f:
+      # create an empty list to store lines in files.
+      df = []
+      line_number = 0
+      # iterate through each line in the input file
+      for num, line in enumerate(f):
+        # If the number of lines iterated through is equal or less than the shard size:
+        if (shard_size is not None) and ((num + 1) - line_number <=
+                                         (shard_size * 4)):
+          # append to list
+          df.append(line)
+        else:
+          # else yield the list
+          shard_num += 1
+          logger.info("Loading shard %d of size %s." %
+                      (shard_num, str(shard_size)))
+          # set the line_number variable to the last line number (num) before 'yield' was called
+          line_number = num
+          # yield list (shard/batch)
+          yield df
+          # Re-initialize list with the index line to begin a new shard.
+          df = [line]
+      if len(df) > 0:
+        yield df
+
+
+class FASTQLoader(DataLoader):
```

Comment:
Can you add to the docs at https://raw.githubusercontent.com/deepchem/deepchem/master/docs/source/api_reference/data.rst?